### PR TITLE
drivers: wifi: Add RPU recovery info

### PR DESCRIPTION
**Pull request by kapbh**
Add RPU recovery information in wifi utils.
It helps to debug watchdog recovery.

### CODE REVIEW

**Review by rlubos — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

File: drivers/wifi/nrfwifi/src/net_if.c [hunk 1]
```diff
@@ -158,6 +159,7 @@ static void nrf_wifi_rpu_recovery_work_handler(struct k_work *work)
 	}
 	rpu_ctx_zep->rpu_recovery_in_progress = false;
 	rpu_ctx_zep->last_rpu_recovery_time_ms = k_uptime_get();
+	rpu_ctx_zep->rpu_recovery_success++;
```
Comment (krish2718) [inline drivers/wifi/nrfwifi/src/net_if.c:162]: Even for failure we are continuing above #148, so, need to adjust this counter only for successful case.

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: drivers/wifi/nrf_wifi/src/wifi_util.c [hunk 2]
```diff
@@ -1013,6 +1056,12 @@ SHELL_STATIC_SUBCMD_SET_CREATE(
 		      nrf_wifi_util_trigger_rpu_recovery,
 		      1,
 		      0),
+	SHELL_CMD_ARG(rpu_recovery_info,
+		      NULL,
+		      "RPU recovery information",
```
Comment (krish2718) [inline drivers/wifi/nrf_wifi/src/wifi_util.c:1061]:
```suggestion
		      "Dump RPU recovery information",
```

File: drivers/wifi/nrf_wifi/src/wifi_util.c [hunk 3]
```diff
@@ -915,6 +915,49 @@ static int nrf_wifi_util_trigger_rpu_recovery(const struct shell *sh,
...
+	fmac_dev_ctx = ctx->rpu_ctx;
+	hal_dev_ctx = fmac_dev_ctx->hal_dev_ctx;
+
+	shell_fprintf(sh,
+		      SHELL_INFO,
+		      "wdt_irq_received: %d\n"
+		      "wdt_irq_ignored: %d\n"
+		      "last_wakeup_now_asserted_time_ms: %lu\n"
+		      "last_wakeup_now_deasserted_time_ms: %lu\n"
+		      "last_rpu_sleep_opp_time_ms: %lu\n"
```
Comment (krish2718) [inline drivers/wifi/nrf_wifi/src/wifi_util.c:945]: do these make sense without the current time?

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

File: drivers/wifi/nrf_wifi/src/wifi_util.c (hunk 3 above)
Comment (kapbh) [reply drivers/wifi/nrf_wifi/src/wifi_util.c]: Adding current time now. we can add some more debug stats later.

**Review by sachinthegreen — APPROVED** [review]

**Review by krish2718 — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**